### ISSUE TITLE
False positive for `tuple[Never] = (1,)`

### ISSUE THREAD

**Issue by jorenham** [microsoft/pyright#8237]
Since the (variadic) type parameters of `tuple` are covariant (or at least, that's how they behave), and `Never` is the "bottom-type", I believe that `tuple[Never]` should *never* accept any tuple (except for some `tuple[Never]`, but that'll have to wait until Python 4 is released).

```python
from typing import Never

never_one: Never = 1  # Type "Literal[1]" is incompatible with type "Never"  (reportAssignmentType)

nevermore: tuple[Never] = (1,)  # passes without error
reveal_type(nevermore)  # Type of "nevermore" is "tuple[Literal[1]]"
```

This is an issue with `1.1.369`, doesn't appear to be new, and is does not depend on any of the config options (that I've tried).
See https://pyright-play.net/?strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoByApgG5EgBQFKp5A%2BmDQFyG0hQC8UAjFTWSAi4iLGAFcEAGyIBtYgIC6nKAApuAGgCUFELQCGkuvAREV-ckN3agA

**Comment by jorenham** [comment]
I know this is a sensitive topic, and I mean no offense, but it might be useful information so I'll say it anyway (but just to be sure, in very small font): 
<sub>mypy correctly rejects the `nevermore` assignment in this example</sub>

**Comment by erictraut** [comment]
Thanks, this is definitely a bug. Pyright is violating the typing spec here. Tuples require a bunch of special-case handling, and there's probably just a hole in that special-case logic. I'll investigate further.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.370
